src/sidebar/util/test/observer-element-size-test.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+const observeElementSize = require('../observe-element-size');
+
+/**
+ * Give MutationObserver, ResizeObserver etc. a chance to deliver their
+ * notifications.
+ */
+function waitForObservations() {
+  return new Promise(resolve => setTimeout(resolve, 1));
+}
+
+describe('observeElementSize', () => {
+  let content;
+  let sizeChanged;
+  let stopObserving;
+
+  beforeEach(() => {
+    sizeChanged = sinon.stub();
+    content = document.createElement('div');
+    content.innerHTML = '<p>Some test content</p>';
+    document.body.appendChild(content);
+  });
+
+  afterEach(() => {
+    stopObserving();
+    content.remove();
+  });
+
+  function startObserving() {
+    stopObserving = observeElementSize(content, sizeChanged);
+  }
+
+  context('when `ResizeObserver` is available', function() {
+    if (typeof ResizeObserver === 'undefined') {
+      this.skip();
+    }
+
+    it('notifies when the element size changes', async () => {
+      startObserving();
+
+      content.innerHTML = '<p>different content</p>';
+      await waitForObservations();
+      assert.called(sizeChanged);
+
+      stopObserving();
+      sizeChanged.reset();
+
+      content.innerHTML = '<p>other content</p>';
+      await waitForObservations();
+      assert.notCalled(sizeChanged);
+    });
+  });
+
+  context('when `ResizeObserver` is not available', () => {
+    let origResizeObserver;
+    beforeEach(() => {
+      origResizeObserver = window.ResizeObserver;
+      window.ResizeObserver = undefined;
+    });
+
+    afterEach(() => {
+      window.ResizeObserver = origResizeObserver;
+    });
+
+    [
+      {
+        description: 'media loads inside the element',
+        triggerCheck: () =>
+          content.dispatchEvent(new Event('load', { bubbles: true })),
+      },
+      {
+        description: 'the window is resized',
+        triggerCheck: () => window.dispatchEvent(new Event('resize')),
+      },
+      {
+        description: "the element's DOM structure changes",
+        triggerCheck: () => (content.innerHTML += '<p>more content</p>'),
+      },
+    ].forEach(({ description, triggerCheck }) => {
+      it(`checks for changes when ${description}`, async () => {
+        startObserving();
+
+        // Change the content height, which is not directly observed.
```

Comment:
"Content height" is not directly observed—because `clientHeight` is what's being observed? I admit to being a little muddled between this and `minHeight` and...anyway...can you help me understand here? Is it basically that an initial `height` of some sort needs to be set so that the observer can see a change? Sorry I'm not following!

Reply:
In the fallback case, the utility monitors the DOM _structure_ but cannot directly monitor the _laid out DOM elements_, which is what is changed by modifying the element's style. This is a difference from when `ResizeObserver` is available, which can monitor all causes of element size changes.